elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/schema/Schema.java

```diff
@@ -94,19 +95,35 @@ public Schema(Class<?> cls, EntityDictionary entityDictionary) {
     }
 
     /**
-     * Finds the {@link Dimension} by name.
+     * Finds the {@link DimensionColumn} by name.
      *
-     * @param dimensionName  The entity field name associated with the searched {@link Dimension}
+     * @param dimensionName  The entity field name associated with the searched {@link DimensionColumn}
      *
-     * @return {@link Dimension} found or {@code null} if not found
+     * @return {@link DimensionColumn} found or {@code null} if not found
      */
-    public Dimension getDimension(String dimensionName) {
+    public DimensionColumn getDimension(String dimensionName) {
         return getDimensions().stream()
                 .filter(dimension -> dimension.getName().equals(dimensionName))
                 .findAny()
                 .orElse(null);
     }
 
+    /**
+     * Finds the {@link TimeDimensionColumn} by name.
+     *
+     * @param dimensionName  The entity field name associated with the searched {@link TimeDimensionColumn}
+     *
+     * @return {@link TimeDimensionColumn} found or {@code null} if not found
+     */
+    public TimeDimensionColumn getTimeDimension(String dimensionName) {
+        return getDimensions().stream()
+                .filter(dimension -> dimension.getName().equals(dimensionName))
```

Comment:
I think we are doing this `stream().filter(dimension -> dimension.getName().equals(dimensionName))` so many times, should we store this in a **name -> dimension** map?

Reply:
👍